Problem with custom genesis file after the latest PR

#### System information

Geth version: `latest develop`
OS & Version: Archlinux
Commit hash : develop

#### Expected behaviour
Load the custom genesis json and launch geth

#### Actual behaviour
Panic

#### Steps to reproduce the behaviour

Private testnet with custom genesis file.
```
geth --networkid=123 --nodiscover --maxpeers=0 --genesis=./genesis_block.json --datadir ./data  --ipcpath="/home/lefteris/.ethereum/geth.ipc" console
```

Here is the custom genesis file:

```js
{
  "config": {
    "homesteadBlock": "0"
  },
  "mixhash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "difficulty": "0x0000000000000000000000000000000000000000000000000000000000000001",
  "gasLimit": "0x47e7c4",
  "extraData": "0x0",
  "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "timestamp": "0x0",
  "alloc": {
    "0x75715927d21042c8f0cbd074b1c6207bcbb482b8": {
      "balance": "133700000000000000000000000000000000"
    },
    "0x452d98daaf4a1f41366958eb9a2401be337662ed": {
      "balance": "133700000000000000000000000000000000"
    },
    "0x9c69a0a394c0045ac37736aa37244fb5cd377022": {
      "balance": "133700000000000000000000000000000000"
    },
    "0x5473f6228e2e14b4c3d7c7342199f7acacbe3e2b": {
      "balance": "133700000000000000000000000000000000"
    },
    "0x503495fa36c94e96823ce50051b2e8e8658cd4e4": {
      "balance": "133700000000000000000000000000000000"
    }
  },
  "nonce": "0xdeadbeefdeadbeef"
}
```

Even if I remove the `config` section I still get the panic.

#### Backtrace

```
I0401 18:28:41.458502 ethdb/database.go:82] Alloted 128MB cache and 1024 file handles to data/chaindata
I0401 18:28:41.461800 ethdb/database.go:169] closed db: data/chaindata
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x535875]

goroutine 1 [running]:
panic(0xe0c320, 0xc820012160)
        /usr/lib/go/src/runtime/panic.go:464 +0x3e6
github.com/ethereum/go-ethereum/core/types.(*Block).Hash(0x0, 0x0, 0x0, 0x0, 0x0)
        /home/lefteris/w/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/types/block.go:388 +0x185
github.com/ethereum/go-ethereum/cmd/utils.MustMakeChainConfig(0xc82027a280, 0x0)
        /home/lefteris/w/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/utils/flags.go:775 +0x10c
github.com/ethereum/go-ethereum/cmd/utils.MakeSystemNode(0xf45c58, 0x4, 0xc8202390a0, 0x17, 0xc820239be0, 0x16, 0x16, 0xc82027a280, 0x5)
        /home/lefteris/w/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/utils/flags.go:663 +0xaf5
main.console(0xc82027a280)
        /home/lefteris/w/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:460 +0xa2
github.com/codegangsta/cli.Command.Run(0xf49ef8, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1089160, 0x30, 0x0, ...)
        /home/lefteris/w/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:174 +0x1399
github.com/codegangsta/cli.(*App).Run(0xc82027a000, 0xc82000a090, 0x9, 0x9, 0x0, 0x0)
        /home/lefteris/w/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:187 +0x112b
main.main()
        /home/lefteris/w/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:373 +0x53
```

--- a/cmd/geth/main.go
+++ b/cmd/geth/main.go
@@ -359,6 +359,12 @@ JavaScript API. See https://github.com/ethereum/go-ethereum/wiki/Javascipt-Conso
 		go metrics.CollectProcessMetrics(3 * time.Second)
 
 		utils.SetupNetwork(ctx)
+
+		// Deprecation warning.
+		if ctx.GlobalIsSet(utils.GenesisFileFlag.Name) {
+			common.PrintDepricationWarning("--genesis is deprecated. Switch to use 'geth init /path/to/file'")
+		}
+
 		return nil
 	}
 

--- a/cmd/utils/flags.go
+++ b/cmd/utils/flags.go
@@ -772,23 +772,22 @@ func MustMakeChainConfig(ctx *cli.Context) *core.ChainConfig {
 	)
 	defer db.Close()
 
-	chainConfig, err := core.GetChainConfig(db, genesis.Hash())
-	if err != nil {
-		if err != core.ChainConfigNotFoundErr {
+	if genesis != nil {
+		// Exsting genesis block, use stored config if available.
+		storedConfig, err := core.GetChainConfig(db, genesis.Hash())
+		if err == nil {
+			return storedConfig
+		} else if err != core.ChainConfigNotFoundErr {
 			Fatalf("Could not make chain configuration: %v", err)
 		}
-		var homesteadBlockNo *big.Int
-		if ctx.GlobalBool(TestNetFlag.Name) {
-			homesteadBlockNo = params.TestNetHomesteadBlock
-		} else {
-			homesteadBlockNo = params.MainNetHomesteadBlock
-		}
-
-		chainConfig = &core.ChainConfig{
-			HomesteadBlock: homesteadBlockNo,
-		}
 	}
-	return chainConfig
+	var homesteadBlockNo *big.Int
+	if ctx.GlobalBool(TestNetFlag.Name) {
+		homesteadBlockNo = params.TestNetHomesteadBlock
+	} else {
+		homesteadBlockNo = params.MainNetHomesteadBlock
+	}
+	return &core.ChainConfig{HomesteadBlock: homesteadBlockNo}
 }
 
 // MakeChainDatabase open an LevelDB using the flags passed to the client and will hard crash if it fails.

--- a/eth/backend.go
+++ b/eth/backend.go
@@ -169,10 +169,6 @@ func New(ctx *node.ServiceContext, config *Config) (*Ethereum, error) {
 
 	// Load up any custom genesis block if requested
 	if len(config.Genesis) > 0 {
-		// Using println instead of glog to make sure it **always** displays regardless of
-		// verbosity settings.
-		common.PrintDepricationWarning("--genesis is deprecated. Switch to use 'geth init /path/to/file'")
-
 		block, err := core.WriteGenesisBlock(chainDb, strings.NewReader(config.Genesis))
 		if err != nil {
 			return nil, err

